File: server/src/main/java/com/example/server/entity/Project.java
package com.example.server.entity;

import com.example.server.entity.enums.ProjectStatus;
import com.fasterxml.jackson.annotation.*;
import lombok.Data;
import lombok.EqualsAndHashCode;
import lombok.ToString;
import org.springframework.data.annotation.Id;
import org.springframework.data.elasticsearch.annotations.Document;

import java.util.List;
import java.util.UUID;

@Data
@ToString(exclude = {"participants", "owner"})
@EqualsAndHashCode
@Document(indexName = "project", shards = 1)
public class Project {
    @Id
    private UUID id;
    private String name;
    private List<ProfArea> profArea;
    private List<User> participants;
    private User owner;
    private List<Tag> tags;
    private String description;
    private ProjectStatus projectStatus;
    private String city;
}

File: client/src/app/table-classes/project.ts
import {ProfArea} from "./prof-area";
import {User} from "./user";
import {ProjectStatus} from "./project-status";
import {Tag} from "./tag";

export class Project {
  constructor(public id: string, public name: string, public profArea: ProfArea[], public participants: User[],
              public owner: User, public tags: Tag[], public description: string,
              public projectStatus: ProjectStatus) {
  }

  getAllProfAreas(){
    let profAreas = "";
    for(var i = 0; i < this.profArea.length; i++){
      profAreas = profAreas + this.profArea[i].name;
    }
    return profAreas.toString;
  }

  public getAllTags(){
    let tags = "";
    for(var i = 0; i < this.tags.length; i++){
      tags = tags + this.tags[i].name;
    }
    return tags.toString;
  }
}
